woocommerce/woocommerce-ios
# [Mobile Payments] Extended IPP onboarding tracks events

Closes: #7525 


### Description


We added a new IPP onboarding step to enable Cash on Delivery in a merchants store, to allow them to take payments for web orders using their card reader.

During that work, we added some new generic onboarding tracks events: this PR extends those so that they are logged by all the previously existing onboarding steps, where appropriate.

Specifically, these events are added/updated:
`card_present_onboarding_learn_more_tapped`: `reason` added.
`card_present_onboarding_step_skipped`: tracked when the merchant skips the `Pending Requirements` screen.
`card_present_onboarding_cta_tapped`: new event, tracked whenever a primary button is tapped on an onboarding screen, including the `reason`.

N.B. for Pending requirements, the Skip button is also the CTA. In this case, we track both events.


### Testing instructions


Hardcode the onboarding step desired in `CardPresentPaymentsOnboardingUseCase.accountChecks(plugin:)`, e.g.

```
    func accountChecks(plugin: CardPresentPaymentsPlugin) -> CardPresentPaymentOnboardingState {
        return .stripeAccountPendingRequirement(plugin: plugin, deadline: Date())
        [...]
```

Launch the app, and go to `Menu > Payments > Continue Setup`
Tap on the button you're looking to test: `Learn More`, `Skip`, or the CTA.
Check the logs for the record of the analytics that were sent, e.g.

```
🔵 Tracked card_present_onboarding_cta_tapped, properties: [AnyHashable("country"): "US", AnyHashable("is_wpcom_store"): false, AnyHashable("reason"): "account_pending_requirements", AnyHashable("blog_id"): 197448330]
```

Also check in the Tracks Live View for the analytics you were expecting to be tracked:

<img width="635" alt="CleanShot 2022-08-26 at 11 43 27@2x" src="https://user-images.githubusercontent.com/2472348/186887179-0b8f357c-9f69-49e3-99b7-7ab89980520e.png">

---
